mne/gui/_vol_stc.py

```diff
@@ -0,0 +1,790 @@
+            self._scalar_bar.SetHeight(0.6)
+            self._scalar_bar.SetWidth(0.05)
+            self._scalar_bar.SetPosition(0.02, 0.2)
+
+        self._update_cmap()  # must be called for volume to render properly
+
+        if show:
+            self.show()
+
+    def _get_src_coord(self):
+        """Get the current slice transformed to source space."""
+        return tuple(np.round(_coord_to_coord(
+            self._current_slice, self._vox_scan_ras_t,
+            self._src_scan_ras_vox_t)).astype(int))
+
+    def _pick_stc_epoch(self):
+        """Select the source time course epoch based on the parameters."""
+        if self._epoch_idx == 'Average':
+            if self._data.dtype == np.int16:
+                stc_data = self._data.mean(axis=0).astype(np.int16)
+            else:
+                stc_data = self._data.mean(axis=0)
+        elif self._epoch_idx == 'Average Power':
+            if self._data.dtype == COMPLEX_DTYPE:
+                stc_data = np.sum(_int_complex_conj(
+                    self._data) // self._data.shape[0], axis=0)
+            else:
+                stc_data = (self._data * self._data.conj()).real.mean(axis=0)
+        elif self._epoch_idx == 'ITC':
+            if self._data.dtype == COMPLEX_DTYPE:
+                stc_data = self._data['re'].astype(np.complex64) + \
+                    1j * self._data['im']
+                stc_data = np.abs((stc_data / np.abs(stc_data)).mean(
+                    axis=0))
+            else:
+                stc_data = np.abs((self._data / np.abs(self._data)).mean(
+                    axis=0))
+        else:
+            stc_data = self._data[int(self._epoch_idx.replace('Epoch ', ''))]
+            if self._data.dtype == COMPLEX_DTYPE:
+                stc_data = _int_complex_conj(stc_data)
+            elif self._is_complex:
+                stc_data = (stc_data * stc_data.conj()).real
+        # do baseline correction
+        if self._baseline != 'None' and self._epoch_idx != 'ITC':
+            if self._data.dtype in (COMPLEX_DTYPE, np.int16):
+                stc_mean = np.sum(stc_data // stc_data.shape[0], axis=0)
+            else:
+                stc_mean = stc_data.mean(axis=-1, keepdims=True)
+            if self._baseline == 'Gain':
+                if self._data.dtype in (COMPLEX_DTYPE, np.int16):
+                    stc_data = stc_data // stc_mean
+                else:
+                    stc_data = stc_data / stc_mean
+            else:
+                assert self._baseline == 'Subtraction'
+                if self._data.dtype in (COMPLEX_DTYPE, np.int16):
+                    neg_mask = stc_data < (-RANGE_VALUE + stc_mean)
+                    pos_mask = stc_data > (RANGE_VALUE + 1 + stc_mean)
+                    stc_data -= stc_mean
+                    stc_data[neg_mask] = -RANGE_VALUE
+                    stc_data[pos_mask] = RANGE_VALUE - 1
+                else:
+                    stc_data -= stc_mean
```

Comment:
same remark we have some shared functions to compute baseline in different ways. I would try to rely on these has much as possible

Reply:
I looked into these, they require a time range and I just hadn't implemented it yet. I should just add a tmin and tmax for the baseline